src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -3,22 +3,55 @@
 /*
  * Created on 05.09.2014
  */
+
+import java.util.Arrays;
+
 /**
  * @author 
  */
 public class UTF8Converter {
+	public static void main(String[] args) {
+		byte[] utf8Bytes = {(byte) 0xF0, (byte) 0x9F, (byte) 0x98, (byte) 0xA5};
+		int codePoint = UTFtoCodePoint(utf8Bytes);
+		System.out.println(Integer.toHexString(codePoint));
+	}
 	public static byte[] codePointToUTF(int x) {
-		byte[] b = null;
-		// TODO allocate b in the right size (depending on x) and fill it with the
-		// UTF-8 encoding of code point x. b[0] shall contain the first byte.
+
+		int len = 4;
+		if (x < 1 << 7) len = 1;
+		else if (x < 1 << 11) len = 2;
+		else if (x < 1 << 16) len = 3;
+		if(x >= 1 << 22) throw new IllegalArgumentException();
+
+		byte[] b = new byte[len];
+		for (int i = len - 1; i > 0; i--) {
+			b[i] = (byte) (0b1000_0000 | (x & 0b0011_1111));
+			x >>= 6;
+		}
+
+		b[0] = (byte) x;
+
+		if(len > 1){
+			for(int i = 0; i < len; i++){
+				b[0] |= (byte) (1 << 8 - len + i);
+			}
+		}
 		return b;
 	}
 
 	public static int UTFtoCodePoint(byte[] bytes) {
 		if (isValidUTF8(bytes)) {
-			// TODO replace return statement below by code to return the code point
-			// UTF-8 encoded in array bytes. bytes[0] contains the first byte
-			return 0;
+			int len = bytes.length;
+			if (len == 1) return bytes[0];
+			int x = 0;
+			x |= bytes[0];
```

Comment:
Das ist vielleicht doch etwas umständlich formuliert.
```suggestion
			int x = bytes[0];
```